src/components/ParameterTabsComponent.js

```diff
@@ -27,34 +29,96 @@ export default function ParameterTabsComponent() {
   const { userInput } = useContext(UserInputContext);
   const { setRIVResults } = useContext(RIVResultContext);
   const { setNotificationStatus } = useContext(NotificationContext);
+  const { selectedReittiviiva } = useContext(SelectedReittiviivaContext);
 
   const handleTabChange = (event, newValue) => {
     setValue(newValue);
   };
   const { setWayareaPolygons } = useContext(WayareaPolygonContext);
+  const { selectedCalculationType } = useContext(
+    SelectedCalculationTypeContext
+  );
 
   const fetchRiskValue = async (values) => {
-    const path = "fairway/calculate_risk";
-    const path_wayarea = "wayarea";
+    let path = "",
+      path_wayarea = "",
+      path_navigationline = "",
+      path_wayarea_navigationline = "",
+      path_reittiviiva = "",
+      path_wayarea_reittiviiva = "";
+
+    if (selectedCalculationType == "reittiviiva") {
+      path = `reittiviiva/calculate_risk?routename=${encodeURIComponent(
+        selectedReittiviiva
+      )}`;
+      path_wayarea = `reittiviiva/wayarea_polygons?routename=${encodeURIComponent(
+        selectedReittiviiva
+      )}`;
+    } else if (selectedCalculationType == "navigationline") {
+      path = `fairway/calculate_risk?vaylat=${encodeURIComponent(
+        values.vaylat
+      )}`;
+      path_wayarea = `wayarea?vaylat=${encodeURIComponent(values.vaylat)}`;
+    } else if (selectedCalculationType == "compare") {
+      path_navigationline = `fairway/calculate_risk?vaylat=${encodeURIComponent(
+        values.vaylat
+      )}`;
+      path_wayarea_navigationline = `wayarea?vaylat=${encodeURIComponent(
+        values.vaylat
+      )}`;
+      path_reittiviiva = `reittiviiva/calculate_risk?routename=${encodeURIComponent(
+        selectedReittiviiva
+      )}`;
+      path_wayarea_reittiviiva = `reittiviiva/wayarea_polygons?routename=${encodeURIComponent(
+        selectedReittiviiva
+      )}`;
+    }
+
```

Comment:
This could be separated into its own function

Reply:
now there is component to generate path

Reply:
Could it be changed to just a function inside the ParameterTabsComponent, as it only concerns with logic for that component? Seeing as the path creation logic doesn't return any JSX, it shouldn't be in a separate component file. If you want to put it in another file, it could be under the /utils folder in an own file.

Reply:
yes i tried now